### PR TITLE
Timeline chart: Update chart after collapsing or expanding entries in…

### PR DESCRIPTION
**Pull request by smouss**
… the tree

Added a new condition in time-graph-chart.ts that allows to fetch new
data when collapsing or expanding entries. The function updateChart()
allows to update the chart's rows simulteanously when it is called
in the timegraph-output-component.ts in theia-trace-extension.
++ Added optional parameter to maybeFetchNewData() that enables the
chart update.

Signed-off-by: Soukaina Moussaoui <soukaina.moussaoui@polymtl.ca>

### CODE REVIEW

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 1]
```diff
@@ -157,6 +162,7 @@ export class TimeGraphChart extends TimeGraphChartLayer {
                 this.addRows(this.rows, this.rowController.rowHeight);
             }
             this.fetching = false;
+            updateChartEnabled=false;
```
Comment (MatthewKhouzam) [inline timeline-chart/src/layer/time-graph-chart.ts:165]: I think you can remove this

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 2]
```diff
@@ -135,17 +134,23 @@ export class TimeGraphChart extends TimeGraphChartLayer {
...
+        const updateChartEnabled = true;
+        this.maybeFetchNewData(updateChartEnabled);
+    }
+
     update() {
         this.updateScaleAndPosition();
     }
 
-    protected async maybeFetchNewData() {
+    protected async maybeFetchNewData(updateChartEnabled?: boolean) {
```
Comment (MatthewKhouzam) [inline timeline-chart/src/layer/time-graph-chart.ts:146]: Call it just update, maybe? not sure!

**Comment by mad-season** [comment]
@MatthewKhouzam @smouss 

Just a heads up. This merge undid the fix for the orphaned state in time-graph-chart.ts selectRowElement.

**Comment by MatthewKhouzam** [comment]
Great catch!
________________________________

**Comment by smouss** [comment]
> @MatthewKhouzam @smouss
> 
> Just a heads up. This merge undid the fix for the orphaned state in time-graph-chart.ts selectRowElement.

Hello! What was the fix about exactly? Thank you.

**Pull request merged**